src/keygen/keygen_commit.rs

```diff
@@ -36,7 +36,6 @@ pub(crate) struct KeygenDecommit {
     ///`sid` corresponds to a unique session identifier.
     pub sid: Identifier,
     pub sender: ParticipantIdentifier,
-    pub chain_code: [u8; 32],
```

Comment:
The `chain_code` variable should still be in the `KeyGen` protocol. Apologies for not being clear on this but I think it makes the most sense to have basically two separate key generation protocols: one for the n out of n case and one for the t out of n case which is the `TShare` protocol. Both of these protocols should support generating HD wallets. If we want to support conversions later (between n out of n keys and t out of n keys) we can make parts of the `TShare` protocol reusable for those contexts in some intelligent way.